src/Sentry.Unity/SentryUnitySdk.cs

```diff
@@ -137,4 +125,77 @@ public void CaptureFeedback(string message, string? email, string? name, bool ad
 
         Sentry.SentrySdk.CurrentHub.CaptureFeedback(message, email, name, hint: hint);
     }
+
+    internal static void HandleWindowsPlayer(SentryUnitySdk unitySdk, SentryUnityOptions options)
+    {
+        // On Windows-Standalone, we disable cache dir in case multiple app instances run over the same path.
+        // Note: we cannot use a named Mutex, because Unity doesn't support it. Instead, we create a file with `FileShare.None`.
+        // https://forum.unity.com/threads/unsupported-internal-call-for-il2cpp-mutex-createmutex_internal-named-mutexes-are-not-supported.387334/
+        if (ApplicationAdapter.Instance.Platform is not RuntimePlatform.WindowsPlayer ||
+            options.CacheDirectoryPath is null)
+        {
+            return;
+        }
+
+        try
+        {
+            unitySdk._lockFile = new FileStream(Path.Combine(options.CacheDirectoryPath, "sentry-unity.lock"), FileMode.OpenOrCreate,
+                FileAccess.ReadWrite, FileShare.None);
+        }
+        catch (Exception ex)
+        {
+            options.DiagnosticLogger?.LogWarning("An exception was thrown while trying to " +
+                                                 "acquire a lockfile on the config directory: .NET event cache will be disabled.", ex);
+            options.CacheDirectoryPath = null;
+            options.AutoSessionTracking = false;
+        }
+    }
+
+    internal static void HandleLateIntegrations(SentryUnityOptions options)
+    {
+        if (options.AttachViewHierarchy)
+        {
+            options.AddEventProcessor(new ViewHierarchyEventProcessor(options));
+        }
+        if (options.AttachScreenshot)
+        {
+            options.AddEventProcessor(new ScreenshotEventProcessor(options));
+        }
+
+        if (!ApplicationAdapter.Instance.IsEditor &&
+            (SentryPlatformServices.UnityInfo?.IL2CPP ?? false) &&
+            options.Il2CppLineNumberSupportEnabled)
+        {
+            if (SentryPlatformServices.UnityInfo.Il2CppMethods is not null)
+            {
+                options.AddExceptionProcessor(new UnityIl2CppEventExceptionProcessor(options));
+            }
+            else
+            {
+                options.DiagnosticLogger?.LogWarning("Failed to find required IL2CPP methods - Skipping line number support");
+            }
+        }
+    }
+
+    internal static void HandlePlatformRestrictedOptions(SentryUnityOptions options, ISentryUnityInfo? unityInfo)
```

Comment:
Again method name is off, seems the guard is: `IsKnownPlatform` so how about:

`InitUnknownPlatform`